src/main/java/org/jvnet/hudson/test/RealJenkinsRule.java

```diff
@@ -504,27 +506,31 @@ public static List<String> getJacocoAgentOptions() {
         return new Statement() {
             @Override public void evaluate() throws Throwable {
                 System.out.println("=== Starting " + description);
-                if (war == null) {
-                    war = findJenkinsWar();
-                }
-                if (home.get() != null) {
+                try {
+                    jenkinsOptions("--webroot=" + createTempDirectory("webroot"), "--pluginroot=" + createTempDirectory("pluginroot"));
```

Comment:
Various controller packagings set these options to temp dirs, so I think it is realistic to set them here automatically. Otherwise the fallback is inside `$JENKINS_HOME` which is undesirable.